### ISSUE TITLE
[BUG] - "Out of space" memory errors running PyTorch notebooks on QHub

### ISSUE THREAD

**Issue by magsol** [nebari-dev/nebari#1332]
### OS system and architecture in which you are running QHub

quansight.qhub.dev

### Expected behavior

Be able to run all the [PyTorch training notebooks here](https://github.com/Quansight/cybera-training-2022) (any of notebooks from 15 on) on a large GPU or CPU instance without any issues.

### Actual behavior

After an indeterminate amount of time (as in, I couldn't put a hard limit on when it happens, but after running through 2-3 of the, specifically the PyTorch ones [e.g., 16 and 17]), I would start getting "out of memory" errors. The exact location in the notebook where the error occurred wouldn't always be the same, but the error itself would:

```
OSError: [Errno 28] No space left on device
```

In parallel, the only error @viniciusdc found in k9s was the following:

```
│ notebook [IPKernelApp] ERROR | No such comm target registered: jupyter.widget.control                                                                                  │
│ notebook [IPKernelApp] WARNING | No such comm: f2b54b1b-0e61-4c4e-b02e-1123de09eae3
```

Unclear if this is related, but included just in case.

### How to Reproduce the problem?

Choose a large GPU or CPU instance and run notebooks 15, 16, and 17. May need to re-run them a couple times.

The only change I could find through running `df -h` is the `shm` disk would be completely full (see the first batch of output in "Command Output").

The only hotfix I could find was restarting the entire instance. Restarting the Jupyter kernel had no effect (see the second batch of output in "Command Output"; the only change from the first is that the `shm` disk is no longer at 100% usage).

It's also worth noting: there's a notebook from [a previous computer vision training](https://github.com/Quansight/Sixty-North-CV-ML-Training) that I excluded that involved training a full AlexNet architecture, and this would _consistently_ never work. It would give me the same error no matter what, with the added information: "perhaps you should raise the shared memory limit". This could be a coincidence or a separate issue entirely, but given both errors referenced hitting a memory limit, I thought it worth to include in this ticket.

### Command output

```bash session
$ df -h
Filesystem                   Size  Used Avail Use% Mounted on
overlay                       95G  5.1G   90G   6% /
tmpfs                         64M     0   64M   0% /dev
tmpfs                         16G     0   16G   0% /sys/fs/cgroup
/dev/sda1                     95G  5.1G   90G   6% /etc/hosts
10.3.244.37:/shared/analyst  492G   18G  474G   4% /shared/analyst
10.3.244.37:/shared/cybera   492G   18G  474G   4% /shared/cybera
10.3.244.37:/shared/users    492G   18G  474G   4% /shared/users
10.3.244.37:/home/magsol     492G   18G  474G   4% /home/magsol
shm                           64M   64M     0 100% /dev/shm
10.3.248.181:/users           98G   92G  6.5G  94% /home/conda/users
10.3.248.181:/cybera          98G   92G  6.5G  94% /home/conda/cybera
10.3.248.181:/analyst         98G   92G  6.5G  94% /home/conda/analyst
10.3.248.181:/default         98G   92G  6.5G  94% /home/conda/default
10.3.248.181:/filesystem      98G   92G  6.5G  94% /home/conda/filesystem
10.3.248.181:/magsol          98G   92G  6.5G  94% /home/conda/magsol
tmpfs                         16G     0   16G   0% /proc/acpi
tmpfs                         16G     0   16G   0% /proc/scsi
tmpfs                         16G     0   16G   0% /sys/firmware

$ df -h
Filesystem                   Size  Used Avail Use% Mounted on
overlay                       95G  5.1G   90G   6% /
tmpfs                         64M     0   64M   0% /dev
tmpfs                         16G     0   16G   0% /sys/fs/cgroup
/dev/sda1                     95G  5.1G   90G   6% /etc/hosts
10.3.244.37:/shared/analyst  492G   17G  475G   4% /shared/analyst
10.3.244.37:/shared/cybera   492G   17G  475G   4% /shared/cybera
10.3.244.37:/shared/users    492G   17G  475G   4% /shared/users
10.3.244.37:/home/magsol     492G   17G  475G   4% /home/magsol
shm                           64M   29M   36M  45% /dev/shm
10.3.248.181:/users           98G   92G  6.5G  94% /home/conda/users
10.3.248.181:/cybera          98G   92G  6.5G  94% /home/conda/cybera
10.3.248.181:/analyst         98G   92G  6.5G  94% /home/conda/analyst
10.3.248.181:/default         98G   92G  6.5G  94% /home/conda/default
10.3.248.181:/filesystem      98G   92G  6.5G  94% /home/conda/filesystem
10.3.248.181:/magsol          98G   92G  6.5G  94% /home/conda/magsol
tmpfs                         16G     0   16G   0% /proc/acpi
tmpfs                         16G     0   16G   0% /proc/scsi
tmpfs                         16G     0   16G   0% /sys/firmware
```

### Versions and dependencies used.

QHub 0.4.1 (quansight.qhub.dev)

The "filesystem-cybera" environment / kernel.

**Comment by viniciusdc** [comment]
@costrouc would you mind having a look into this as well?
